beacon_node/beacon_chain/src/schema_change/migration_schema_v24.rs

```diff
@@ -0,0 +1,380 @@
+                            slot,
+                            latest_block_root: old_summary.latest_block_root,
+                            latest_block_slot: old_summary.latest_block_slot,
+                            previous_state_root,
+                            diff_base_state,
+                        };
+                        let op = new_summary.as_kv_store_op(state_root);
+                        // It's not ncessary to immediately commit the summaries of states that are
+                        // ReplayFrom. However we do so for simplicity.
+                        db.hot_db.do_atomically(vec![op])?;
+                    }
+                }
+            }
+
+            // 3. Stage old data for deletion.
+            if slot % T::EthSpec::slots_per_epoch() == 0 {
+                migrate_ops.push(KeyValueStoreOp::DeleteKey(
+                    DBColumn::BeaconState,
+                    state_root.as_slice().to_vec(),
+                ));
+            }
+
+            // Delete previous summaries
+            migrate_ops.push(KeyValueStoreOp::DeleteKey(
+                DBColumn::BeaconStateSummary,
+                state_root.as_slice().to_vec(),
+            ));
+
+            summaries_written += 1;
+            if last_log_time.elapsed() > Duration::from_secs(5) {
+                last_log_time = Instant::now();
+                info!(
+                    diffs_written,
+                    summaries_written,
+                    summaries_count = state_summaries_dag.summaries_count(),
+                    "Hot states migration in progress"
+                );
+            }
+        }
+    }
+
+    // TODO(hdiff): Should run hot DB compaction after deleting potentially a lot of states. Or should wait
+    // for the next finality event?
+    info!(
+        diffs_written,
+        summaries_written,
+        summaries_count = state_summaries_dag.summaries_count(),
+        "Hot states migration complete"
+    );
+
+    Ok(migrate_ops)
+}
+
+pub fn downgrade_from_v24<T: BeaconChainTypes>(
+    db: Arc<HotColdDB<T::EthSpec, T::HotStore, T::ColdStore>>,
+) -> Result<Vec<KeyValueStoreOp>, Error> {
+    let state_summaries = db
+        .load_hot_state_summaries()?
+        .into_iter()
+        .map(|(state_root, summary)| (state_root, summary.into()))
+        .collect::<Vec<(Hash256, DAGStateSummary)>>();
+
+    info!(
+        summaries_count = state_summaries.len(),
+        "DB downgrade of v24 state summaries started"
+    );
+
+    let state_summaries_dag = StateSummariesDAG::new(state_summaries)
+        .map_err(|e| Error::MigrationError(format!("Error on new StateSumariesDAG {e:?}")))?;
+
+    let mut migrate_ops = vec![];
+    let mut states_written = 0;
+    let mut summaries_written = 0;
+    let mut last_log_time = Instant::now();
+
+    // TODO(tree-states): What about the anchor_slot? Is it safe to run the prior version of
+    // Lighthouse with an a higher anchor_slot than expected?
```

Comment:
Yeah it's fine, the `anchor_slot` wasn't used for anything except informational prior to this.

It used to be used, to reject gossip blocks that were earlier than it, but then we removed this because it was redundant with rejecting blocks prior to the split/finalization.